### ISSUE TITLE
Mooshroom mushrooms render incorrectly

### ISSUE THREAD

**Issue by mwolfe25** [sp614x/optifine#3481]
Mooshroom mushrooms render off of the mooshroom geometry

**Comment by HiItsDevin** [comment]
Can you please provide screenshots?

**Comment by mwolfe25** [comment]
![2020-02-29_11 39 20](https://user-images.githubusercontent.com/49489179/75612400-99f36880-5ae8-11ea-9fc2-3b77d194e6c0.png)
![2020-02-29_11 40 43](https://user-images.githubusercontent.com/49489179/75612401-9a8bff00-5ae8-11ea-864b-522671893db1.png)

**Comment by Regnander** [comment]
It would help if you also posted a screenshot with F3, since it gives all lot of info what settings you have activated and which version you are using.

**Comment by Zefferis** [comment]
I am able to reproduce these results on Optifine G1 pre13

Currently in the cem model documentation it states:

```
# Special cases:
# - the mushrooms on the back of the Mooshroom can be customized with the textures 
#   "assets/minecraft/textures/entity/cow/red_mushroom.png"  
#   "assets/minecraft/textures/entity/cow/brown_mushroom.png"  
```

Yet if a mushroom is used it places the mushroom in the incorrect location relative to the model.

![image](https://user-images.githubusercontent.com/15270437/79024941-40477a80-7b39-11ea-9473-3de0a43edec0.png)

Also the mushroom model that is rendered acts differently than the normal mushroom model in that it is flipping the left portion of the texture to the right instead of it being a normal complete texture? I may be using the incorrect words here but the image should clarify:

![image](https://user-images.githubusercontent.com/15270437/79025104-b6e47800-7b39-11ea-9e44-511ce36fcb49.png)

![image](https://user-images.githubusercontent.com/15270437/79025115-bcda5900-7b39-11ea-92d2-a21d3e647271.png)

![image](https://user-images.githubusercontent.com/15270437/79025132-c499fd80-7b39-11ea-8eb4-f5d953af1d40.png)

All 4 corners of the mooshroom's mushroom model have the white/blue pixel, when only 2 should according to the texture itself, it's mirrored left to right.

**Comment by sp614x** [comment]
Fixed in 1.16.2 pre7
